File: src/app/browser/main/tab/page/navigation/history/forward.cpp
#include "forward.hpp"

using namespace app::browser::main::tab::page::navigation::history;

Forward::Forward(
    const Glib::RefPtr<Gio::SimpleAction> & ACTION__FORWARD
) {
    set_action_name(
        "win.main_tab_page_navigation_history_forward" // @TODO
    );

    set_icon_name(
        "go-next-symbolic"
    );

    set_tooltip_text(
        _("Forward")
    );
}

void Forward::refresh(
    const bool & ENABLED
) {
    // @TODO update action status
}

File: src/app/browser/window/tab/item/action/history/forward.rs
use gtk::{gio::SimpleAction, glib::uuid_string_random};

pub trait Forward {
    fn forward() -> Self;
}

impl Forward for SimpleAction {
    fn forward() -> Self {
        let forward = SimpleAction::new(&uuid_string_random(), None);
        forward.set_enabled(false);
        forward
    }
}
